File: backend/src/main/scala/microservices/train/objects/TrainAdminSessionPlannerResponse.scala
package com.typesafe.travel.train.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TrainAdminSessionPlannerResponse(
    managerId: String,
    operatorCode: String,
    email: String,
    displayName: String,
    status: String,
    managedTrains: List[TrainPlannerResponse]
)
object TrainAdminSessionPlannerResponse:
  given sourceEncoder: Encoder[TrainAdminSessionPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[TrainAdminSessionPlannerResponse] = deriveDecoder

File: frontend/src/microservices/operations/train/objects/TrainAdminSessionPlannerResponse.ts
import type { TrainPlannerResponse } from '@/microservices/train/objects/TrainPlannerResponse'

export type TrainAdminSessionPlannerResponse = {
  managerId: string
  operatorCode: string
  email: string
  displayName: string
  status: string
  managedTrains: TrainPlannerResponse[]
}

export const trainAdminSessionPlannerResponseFromJson = (json: string): TrainAdminSessionPlannerResponse =>
  JSON.parse(json) as TrainAdminSessionPlannerResponse

export const trainAdminSessionPlannerResponseToJson = (value: TrainAdminSessionPlannerResponse): string =>
  JSON.stringify(value)

